custom_components/ithodaalderop/sensor.py

```diff
@@ -83,32 +79,30 @@ async def async_setup_entry(
     async_add_entities: AddEntitiesCallback,
 ) -> None:
     """Set up Itho add-on sensors from config entry based on their type."""
+
     sensors = []
     # TODO: This can probably be simplified after refactoring of configflow / itho units
-    if config_entry.data[CONF_CVE_TYPE] == "noncve":
+    if config_entry.data[CONF_ADDON_TYPE] == "noncve":
         for description in NONCVESENSORS:
             description.key = f"{MQTT_BASETOPIC["noncve"]}/{MQTT_STATETOPIC["noncve"]}"
             sensors.append(IthoSensor(description, config_entry, AddOnType.NONCVE))
 
-    if config_entry.data[CONF_CVE_TYPE] == "cve":
+        (sensors.append(IthoSensor(description, config_entry, AddOnType.REMOTES)) for description in _create_remotes(config_entry))
+
+    if config_entry.data[CONF_ADDON_TYPE] == "cve":
         for description in CVESENSORS:
             description.key = f"{MQTT_BASETOPIC["cve"]}/{MQTT_STATETOPIC["cve"]}"
             sensors.append(IthoSensor(description, config_entry, AddOnType.CVE))
 
-    if config_entry.data[CONF_USE_WPU]:
+        (sensors.append(IthoSensor(description, config_entry, AddOnType.REMOTES)) for description in _create_remotes(config_entry))
+
+    if config_entry.data[CONF_ADDON_TYPE] == "wpu":
         for description in WPUSENSORS:
             description.key = f"{MQTT_BASETOPIC["wpu"]}/{MQTT_STATETOPIC["wpu"]}"
             sensors.append(IthoSensor(description, config_entry, AddOnType.WPU))
 
-    if config_entry.data[CONF_USE_REMOTES]:
-        (sensors.append(IthoSensor(description, config_entry, AddOnType.REMOTES)) for description in _create_remotes(config_entry))
-
-    if config_entry.data[CONF_USE_AUTOTEMP]:
-        (sensors.append(IthoSensor(description, config_entry, AddOnType.AUTOTEMP)) for description in _create_autotemprooms(config_entry))
-
     async_add_entities(sensors)
```

Comment:
Don't we miss autotemp now?

Reply:
I broke it when manually merging it with main. Will fix it.